tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -609,6 +609,23 @@ def conditional_promotion(
     assert np.allclose(np.where(mask, a, 10), out)
 
 
+def test_conditional_promotion1(fieldview_backend):
+    size = 10
+    mask = np_as_located_field(IDim)(np.zeros((size,), dtype=bool))
+    mask.array()[0 : (size // 2)] = True
+    out = np_as_located_field(IDim)(np.zeros((size,)))
+
+    @field_operator(backend=fieldview_backend)
+    def conditional_promotion(
+        mask: Field[[IDim], bool],
+    ) -> Field[[IDim], float64]:
+        return where(mask, 5.0, 10.0)
+
+    conditional_promotion(mask, out=out, offset_provider={})
+
+    assert np.allclose(np.where(mask, 5.0, 10), out)
+
+
```

Comment:
Lets add a test for the case with (IDim, JDIm) in the mask and (IDim) in the true/false branches. This probably does not work with the current implementation

Reply:
sure

Reply:
I added that test, and it failed. So I'm reverting back to my original logic for the type promotion. It seems to pass now.